fiftyone/brain/internal/core/pinecone.py

```diff
@@ -243,9 +309,29 @@ def index(self):
     def total_index_size(self):
         if self._index is None:
             return 0
-
         return self._index.describe_index_stats()["total_vector_count"]
 
+    @property
+    def ready(self):
+        if self._pinecone:
+            return self._pinecone.describe_index(
+                self.config.index_name
+            ).status["ready"]
+        return pinecone.describe_index(self.config.index_name).status["ready"]
+
+    def verify_total_index_size(self, expected_size, timeout=10, interval=1):
```

Comment:
It's unfortunate that Pinecone itself doesn't have a concept of what MongoDB calls "write concern" that allows you to wait for their product to reflect the changes that were requested 😢

In practice, when potentially multiple people are working on the same Pinecone index, it feels a bit hacky to wait for the index to reach a certain size. Maybe someone else deleted a sample from the index while this was polling, for example. The ideal thing would be "wait until Pinecone acknowledges that my current transaction completes". But that's not something we should build if Pinecone doesn't support it.

In my opinion, this feels more like a util function that we should put in the `tests` folder to start with so that our integration tests pass (it could be used by any backend, FWIW).

Reply:
If you feel strongly that this is a util we should expose to users, then we can keep it here. Up to you 😄

Reply:
:+1: Moved this to `test_similarity.py`.